epochStart/metachain/systemSCs.go

```diff
@@ -258,39 +276,32 @@ func (s *systemSCProcessor) ProcessSystemSmartContract(
 	}
 
 	if s.flagSwitchJailedWaiting.IsSet() {
-		err := s.computeNumWaitingPerShard(validatorInfos)
+		err := s.computeNumWaitingPerShard(validatorsInfoMap)
 		if err != nil {
 			return err
 		}
 
-		err = s.swapJailedWithWaiting(validatorInfos)
+		err = s.swapJailedWithWaiting(validatorsInfoMap)
 		if err != nil {
 			return err
 		}
 	}
 
 	if s.flagStakingV2Enabled.IsSet() {
-		err := s.prepareRewardsData(validatorInfos)
+		err := s.prepareStakingDataForEligibleNodes(validatorsInfoMap)
 		if err != nil {
 			return err
 		}
 
-		err = s.fillStakingDataForNonEligible(validatorInfos)
+		numUnStaked, err := s.unStakeNonEligibleNodes(validatorsInfoMap, epoch)
 		if err != nil {
 			return err
 		}
 
-		numUnStaked, err := s.unStakeNodesWithNotEnoughFunds(validatorInfos, epoch)
+		err = s.stakeNodesFromQueue(validatorsInfoMap, numUnStaked, nonce, common.NewList)
```

Comment:
flag is missing here

Reply:
stakeNodesFromQueue (numUnstaked) is missing

Reply:
Right, forgot about the flag, added it;
`stakeNodesFromQueue` is called on L301